### PR TITLE
HPCC-19626 imgkey.ecl test fails in roxie regression suite

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Manually stepped through execution of imgkey.ecl

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: rtl/include/eclhelper.hpp [hunk 1]
```diff
@@ -2337,7 +2339,8 @@ struct IHThorIndexReadBaseArg : extends IHThorCompoundBaseArg
     virtual IOutputMetaData * queryDiskRecordSize() = 0;            // Expected layout
     virtual IOutputMetaData * queryProjectedDiskRecordSize() = 0;   // Projected layout
     virtual unsigned getFlags() = 0;
-    virtual unsigned getFormatCrc() = 0;
+    virtual unsigned getFormatCrc() = 0;                             // Corresponding to projectedDiskRecordSize, and should be renamed
```
Comment (ghalliday) [inline rtl/include/eclhelper.hpp]: This should really be renamed at the same time, so you can check that the correct function is being called - by roxie and thor.

File: roxie/ccd/ccdactivities.cpp [hunk 2]
```diff
@@ -2395,7 +2398,8 @@ class CRoxieIndexActivityFactory : public CRoxieKeyedActivityFactory
 
     void init(IHThorIndexReadBaseArg * helper, IPropertyTree &graphNode)
     {
-        formatCrc = helper->getFormatCrc();
+        projectedCrc = helper->getFormatCrc();
+        expectedCrc = helper->getFormatCrc();  // MORE - some time these will be different
```
Comment (ghalliday) [inline roxie/ccd/ccdactivities.cpp]: getDiskFormatCrc?

File: roxie/ccd/ccdfile.cpp [hunk 3]
```diff
@@ -2076,7 +2076,7 @@ class CResolvedFile : implements IResolvedFileCreator, implements ISDSSubscripti
         else
             mb.append(false);
     }
-    virtual ITranslatorSet *getTranslators(int projectedFormatCrc, bool skipFileFormatCrcCheck, IOutputMetaData *projected, IOutputMetaData *expected, RecordTranslationMode mode, bool isIndex) const override
+    virtual ITranslatorSet *getTranslators(int projectedFormatCrc, IOutputMetaData *projected, int expectedFormatCrc, IOutputMetaData *expected, bool skipFileFormatCrcCheck, RecordTranslationMode mode, bool isIndex) const override
```
Comment (ghalliday) [inline roxie/ccd/ccdfile.cpp]: A good change.  I was planning on doing something very similar to the equivalent thor functions when I fixed HPCCC-19590.

File: roxie/ccd/ccdserver.cpp [hunk 4]
```diff
@@ -22769,10 +22771,11 @@ class CRoxieServerBaseIndexActivityFactory : public CRoxieServerActivityFactory
             indexfile.setown(queryFactory.queryPackage().lookupFileName(indexName, isOpt, true, true, queryFactory.queryWorkUnit(), true));
             if (indexfile)
             {
-                unsigned formatCrc = indexHelper->getFormatCrc();
+                unsigned projectedCrc = indexHelper->getFormatCrc();
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: getProjectedFormatCrc?

File: roxie/ccd/ccdserver.cpp [hunk 5]
```diff
@@ -22859,10 +22862,11 @@ class CRoxieServerIndexActivity : public CRoxieServerActivity, implements IRoxie
         varFileInfo.setown(resolveLFNIndex(indexName, isOpt));
         if (varFileInfo)
         {
-            unsigned formatCrc = indexHelper.getFormatCrc();
+            unsigned projectedCrc = indexHelper.getFormatCrc();
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: getProjected...

Comment (ghalliday) [reply roxie/ccd/ccdserver.cpp]: and following

**Review by ghalliday — COMMENTED** [review]
A few comments/questions.

**Comment by ghalliday** [comment]
@richardkchapman

**Comment by richardkchapman** [comment]
@ghalliday please re-review

**Comment by ghalliday** [comment]
@richardkchapman looks good.  Please squash and rebase.

**Comment by richardkchapman** [comment]
@ghalliday Squashed/rebased. Rebase was not 100% trivial so please scan to make sure I got it right

**Pull request merged**